docs/tutorials/sparse/csr.md

```diff
@@ -0,0 +1,338 @@
+- SciPy - A section of this tutorial uses SciPy package in Python. If you don't have SciPy, the example in that section will be ignored.
+- GPUs - A section of this tutorial uses GPUs. If you don't have GPUs on your machine, simply set the variable `gpu_device` (set in the GPUs section of this tutorial) to `mx.cpu()`.
+
+## Compressed Sparse Row Matrix
+
+A CSRNDArray represents a 2D matrix as three separate 1D arrays: **data**, **indptr** and **indices**, where the column indices for row `i` are stored in `indices[indptr[i]:indptr[i+1]]` in ascending order, and their corresponding values are stored in `data[indptr[i]:indptr[i+1]]`.
+
+- **data**: CSR format data array of the matrix
+- **indices**: CSR format index array of the matrix
+- **indptr**: CSR format index pointer array of the matrix
+
+### Example Matrix Compression
+
+For example, given the matrix:
+```
+[[7, 0, 8, 0]
+ [0, 0, 0, 0]
+ [0, 9, 0, 0]]
+```
+
+We can compress this matrix using CSR, and to do so we need to calculate `data`, `indices`, and `indptr`.
+
+The `data` array holds all the non-zero entries of the matrix in row-major order. Put another way, you create a data array that has all of the zeros removed from the matrix, row by row, storing the numbers in that order. Your result:
+
+    data = [7, 8, 9]
+
+The `indices` array stores the column index for each non-zero element in `data`. As you cycle through the data array, starting with 7, you can see it is in column 0. Then looking at 8, you can see it is in column 2. Lastly 9 is in column 1. Your result:
+
+    indices = [0, 2, 1]
+
+The `indptr` array is what will help identify the rows where the data appears. It stores the index into `data` of the first non-zero element number of each row of the matrix. This array always starts with 0 (reasons can be explored later), so indptr[0] is 0. Each subsequent value in the array is the aggregate number of non-zero elements up to that row. Looking at the first row of the matrix you can see two non-zero values, so indptr[1] is 2. The next row contains all zeros, so the aggregate is still 2, so indptr[2] is 2. Finally, you see the last row contains one non-zero element bring the aggregate to 3, so indptr[3] is 3. Your result:
+
+    indptr = [0, 2, 2, 3]
+
+Note that in MXNet, the column indices for a given row are always sorted in ascending order,
+and duplicated column indices for the same row are not allowed.
+
+## Array Creation
+
+There are a few different ways to create a `CSRNDArray`, but first let's recreate the matrix we just discussed using the `data`, `indices`, and `indptr` we calculated.
+
+* We can create a CSRNDArray with data, indices and indptr by using the `csr_matrix` function:
+
+
+```python
+import mxnet as mx
+# Create a CSRNDArray with python lists
+shape = (3, 4)
+data_list = [7, 8, 9]
+indptr_list = [0, 2, 2, 3]
+indices_list = [0, 2, 1]
+a = mx.nd.sparse.csr_matrix(data_list, indptr_list, indices_list, shape)
```

Comment:
I like this approach. In fact, we should address this towards the beginning by showing how it describes the data and even extend it to show how these values can be used to regenerate the full dataset. 
Also, I think it would help to build up the scenarios a bit more. Using only 0's and 1's to show that duplicated data is fine. (There was some content before that mentioned you couldn't duplicate data, but it was changed to refer to the indices.) Without an explicit example the reader may assume incorrectly that data should be unique. Then, as we move to this example, one or two duplicated numbers would be good as well as a larger dataset. We should have an example that demonstrates the compression factor since the small ones we currently have don't show any savings of space.